CMakeLists.txt

```diff
@@ -195,9 +195,9 @@ zephyr_cc_option(-fno-strict-overflow)
 zephyr_cc_option(-Wno-pointer-sign)
 
 if(CONFIG_STACK_CANARIES)
-  zephyr_cc_option(-fstack-protector-all)
+  zephyr_cc_option_nocheck(-fstack-protector-all)
 else()
-  zephyr_cc_option(-fno-stack-protector)
+  zephyr_cc_option_nocheck(-fno-stack-protector)
```

Comment:
you can get rid of the -fno-stack-protector clause entirely, it's not like its on by default or something.

Reply:
done